b2sdk/session.py

```diff
@@ -30,13 +30,31 @@ def __getattr__(self, name):
         @functools.wraps(f)
         def wrapper(*args, **kwargs):
             auth_failure_encountered = False
+            token_type = getattr(f, 'token_type', TokenType.API)
             # download_by_name uses different URLs
             url_factory = kwargs.pop('url_factory', self._api.account_info.get_api_url)
             while 1:
-                api_url = url_factory()
-                account_auth_token = self._api.account_info.get_account_auth_token()
                 try:
-                    return f(api_url, account_auth_token, *args, **kwargs)
+                    if token_type in [TokenType.UPLOAD_SMALL, TokenType.UPLOAD_PART]:
```

Comment:
use a tuple instead of a list here. Always default to tuple and only use a list if it is supposed to be mutable.